diff --git a/drivers/pci/pcie_dw_rockchip.c b/drivers/pci/pcie_dw_rockchip.c
index dd81acf4f86..bd1bd0eb2df 100644
--- a/drivers/pci/pcie_dw_rockchip.c
+++ b/drivers/pci/pcie_dw_rockchip.c
@@ -44,9 +44,7 @@ DECLARE_GLOBAL_DATA_PTR;
 #define dev_info dev_err
 #endif
 
-#ifdef DEBUG
-#define dev_dbg dev_err
-#else
+#ifndef DEBUG
 #define dev_dbg(dev, fmt, ...)					\
 ({								\
 	if (0)							\
diff --git a/include/common.h b/include/common.h
index f32cd284c3c..db35a4b6729 100644
--- a/include/common.h
+++ b/include/common.h
@@ -39,6 +39,8 @@ typedef volatile unsigned char	vu_char;
 #include <image.h>
 #include <stacktrace.h>
 
+#define DEBUG 1
+
 /* Bring in printf format macros if inttypes.h is included */
 #define __STDC_FORMAT_MACROS
 
